lms/djangoapps/course_api/tests/test_api.py

```diff
@@ -0,0 +1,218 @@
+"""
+Test for course API
+"""
+
+from datetime import datetime
+from django.contrib.auth.models import AnonymousUser
+#from django.http import Http404
+from django.test import RequestFactory
+from rest_framework.exceptions import NotFound, PermissionDenied
+
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase, ModuleStoreTestCase
+from xmodule.course_module import DEFAULT_START_DATE
+
+from ..api import course_detail, list_courses
+from .mixins import CourseApiFactoryMixin
+
+
+class CourseApiTestMixin(CourseApiFactoryMixin):
+    """
+    Establish basic functionality for Course API tests
+    """
+
+    maxDiff = 5000  # long enough to show mismatched dicts
```

Comment:
maxDiff is used by unittest.  It controls the output of failing tests.  These tests have a large diff, so by default, you can't why they fail.  Setting this gives more helpful output.